src/transformers/modeling_utils.py

```diff
@@ -1777,10 +1788,29 @@ def save_pretrained(
 
         # Save the config
         if is_main_process:
-            model_to_save.config.save_pretrained(save_directory)
+            if not model_to_save._hf_peft_config_loaded:
+                model_to_save.config.save_pretrained(save_directory)
             if self.can_generate():
                 model_to_save.generation_config.save_pretrained(save_directory)
 
+        if model_to_save._hf_peft_config_loaded:
```

Comment:
Not sure, but should this be inside of `if is_main_process`?

Reply:
Great catch, indeed, yes!